app/src/components/search/filters/selectFilter.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  Accordion as ChakraAccordion,
+  AccordionButton,
+  AccordionItem,
+  AccordionPanel,
+} from "@chakra-ui/react";
+import {
+  Icon,
+  Link,
+  Radio,
+  Flex,
+  Box,
+  RadioGroup,
+} from "@nypl/design-system-react-components";
+import { useState } from "react";
+
+export type FilterOption = {
+  name: string;
+  count: number;
+};
+
+export type FilterCategory = {
+  name: string;
+  options: FilterOption[];
+};
+
+type SelectFilterProps = {
+  filter: FilterCategory;
+  isOpen: boolean;
+  onToggle: () => void;
+};
+
+const SelectFilter = ({ filter, isOpen, onToggle }: SelectFilterProps) => {
+  const [controller, setController] = useState<AbortController | null>(null);
+  const [selected, setSelected] = useState<string | null>(null);
+
+  const getIcon = (isExpanded: boolean) => {
+    const iconRotation = isExpanded ? "rotate180" : "rotate0";
+    return (
+      <Icon
+        name="arrow"
+        size="small"
+        color="ui.black"
+        iconRotation={iconRotation}
+      />
+    );
+  };
+
+  const modalLink = (facet: string) => {
+    return (
+      <Link
+        sx={{ fontSize: "14px", color: "ui.link" }}
+        isUnderlined={false}
+        href={"add-filter"}
+      >{`View all ${facet.toLowerCase()}${
+        facet === "Publishers" ? `` : `s`
+      }`}</Link>
+    );
+  };
+
+  const onChange = (newSelection: string) => {
+    if (controller) {
+      controller.abort();
+    }
+    const newController = new AbortController();
```

Comment:
I will have to double check but I'm pretty sure AbortController should _only_ be used with `fetch`. I imagined this being used on the server component to cancel requests when multiple ones are made, not necessarily for controlling user events. This might be doable with a boolean value instead of creating an abort controller but I'll verify this.

Reply:
I am finding more info on using the `AbortController` with fetch rather than without. This was probably one of the most helpful https://blog.comendrun.com/enhancing-reactnextjs-app-with-native-abortcontroller-interface-in-javascripttypescript

You're right that you should handle the behavior here rather than in the API handlers, but you can do the same without the `AbortController`. It adds a bit more complexity and is not associated with any API requests.

As an aside, this was also helpful but it's more related to timeouts. This might be helpful, at a later time, for the fetch with timeout function https://dtang.dev/timing-out-fetch-requests/